app/src/main/java/com/github/sdp_begreen/begreen/adapters/ExpandableListAdapter.kt

```diff
@@ -39,12 +41,16 @@ class ExpandableListAdapter(
         return groups[groupPosition]
     }
 
-    // Returns the child (item) object in the specified group and child position.
+    // Returns the randomly chosen child (item) object in the specified group.
     override fun getChild(groupPosition: Int, childPosition: Int): Any {
         if (groupPosition < 0 || groupPosition >= groups.size || childPosition < 0 || childPosition >= groups[groupPosition].items.size) {
             throw IllegalArgumentException()
         }
-        return groups[groupPosition].items[childPosition]
+
+        // generated random
+        val rnds = (0..groups[groupPosition].items.size-1).random()
```

Comment:
This is a really weird design choice. Your new feature is not compatible with your previous implementation! In addition, you did not even modify the API accordingly. What is the point of keeping `childPosition` here for example? At least add a `random` flag, and even this is a rudimentary solution...